pkg/cloudprovider/aws/apis/v1alpha1/tags.go

```diff
@@ -29,13 +29,17 @@ const (
 	KarpenterTagKeyFormat = "karpenter.sh/cluster/%s"
 )
 
-func ManagedTagsFor(clusterName string) map[string]string {
-	// tags to be applied on AWS resources created by Karpenter (instances, launchTemplates..)
-	return map[string]string{
-		"Name": fmt.Sprintf("karpenter.sh/%s", clusterName),
+func ManagedTagsFor(clusterName string, customTags map[string]string) map[string]string {
+	managedTags := map[string]string{
 		fmt.Sprintf(ClusterTagKeyFormat, clusterName):   "owned",
 		fmt.Sprintf(KarpenterTagKeyFormat, clusterName): "owned",
 	}
+	// If the provisioner spec includes the Name tag, honor that value. Otherwise, use the default.
+	if _, exists := customTags["Name"]; !exists {
+		managedTags["Name"] = fmt.Sprintf("karpenter.sh/%s", clusterName)
```

Comment:
Thoughts on setting this to `karpenter.sh/cluster/%s/provisioner/%s`

Reply:
Will do, once I get clarification on the previous question, I'll make this the default.

Reply:
@ellistarn You had mentioned that I could get the provisioner name from the ctx, but I only see the following in the options. Tried looking for occurrences of getting the provisioner name from the ctx, but couldn't find any.  Do you have a snippet of getting the provisioner from the ctx?
```
type Options struct {
	ClusterName     string
	ClusterEndpoint string
	MetricsPort     int
	HealthProbePort int
	WebhookPort     int
	KubeClientQPS   int
	KubeClientBurst int
}
```

Reply:
@ellistarn done and tested